doc/src/manual/constructors.md

```diff
@@ -421,56 +415,71 @@ defining sophisticated behavior is typically quite simple.
 ## Case Study: Rational
 
 Perhaps the best way to tie all these pieces together is to present a real world example of a
-parametric composite type and its constructor methods. To that end, here is beginning of [rational.jl](https://github.com/JuliaLang/julia/blob/master/base/rational.jl),
+parametric composite type and its constructor methods. To that end, here is the (slightly modified) beginning of [rational.jl](https://github.com/JuliaLang/julia/blob/master/base/rational.jl),
 which implements Julia's [Rational Numbers](@ref):
 
-```julia
-immutable Rational{T<:Integer} <: Real
-    num::T
-    den::T
-
-    function Rational(num::T, den::T)
-        if num == 0 && den == 0
-            error("invalid rational: 0//0")
-        end
-        g = gcd(den, num)
-        num = div(num, g)
-        den = div(den, g)
-        new(num, den)
-    end
-end
-Rational{T<:Integer}(n::T, d::T) = Rational{T}(n,d)
-Rational(n::Integer, d::Integer) = Rational(promote(n,d)...)
-Rational(n::Integer) = Rational(n,one(n))
-
-//(n::Integer, d::Integer) = Rational(n,d)
-//(x::Rational, y::Integer) = x.num // (x.den*y)
-//(x::Integer, y::Rational) = (x*y.den) // y.num
-//(x::Complex, y::Real) = complex(real(x)//y, imag(x)//y)
-//(x::Real, y::Complex) = x*y'//real(y*y')
-
-function //(x::Complex, y::Complex)
-    xy = x*y'
-    yy = real(y*y')
-    complex(real(xy)//yy, imag(xy)//yy)
-end
+```jldoctest rational
+julia> immutable OurRational{T<:Integer} <: Real
+           num::T
+           den::T
+           function OurRational(num::T, den::T)
+               if num == 0 && den == 0
+                    error("invalid rational: 0//0")
+               end
+               g = gcd(den, num)
+               num = div(num, g)
+               den = div(den, g)
+               new(num, den)
+           end
+       end
+
+julia> OurRational{T<:Integer}(n::T, d::T) = OurRational{T}(n,d)
+OurRational
+
+julia> OurRational(n::Integer, d::Integer) = OurRational(promote(n,d)...)
+OurRational
+
+julia> OurRational(n::Integer) = OurRational(n,one(n))
+OurRational
+
+julia> //(n::Integer, d::Integer) = OurRational(n,d)
+// (generic function with 1 method)
```

Comment:
that might work, but should really show people how to write code and extend operators outside of Base, since working within Base isn't something most people are going to do for a while (if at all)